client/schema_field_description.go

```diff
@@ -10,10 +10,36 @@
 
 package client
 
-import "fmt"
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
 
 // FieldKind describes the type of a field.
-type FieldKind uint8
+type FieldKind interface {
+	// String returns the string representation of this FieldKind.
+	String() string
+
+	// Underlying returns the unterlying Kind as a string.
+	//
+	// If this is an array, it will return the element kind, else it will return the same as
+	// [String()].
```

Comment:
thought: I think that this notation is used in some other places as well but here it threw me off given the context. I was wondering why calling `Undelying()` on a non array would return `String()` wrapped in an array.

Reply:
Square brackets are how to correctly link stuff in Go docs, although I'm not really sure it works for funcs.  I used to use backticks as that feels more normal (and is what rust uses), but the Go devs went with square brackets in 1.19:

https://tip.golang.org/doc/comment#links